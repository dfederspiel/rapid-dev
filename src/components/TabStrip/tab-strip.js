import React, { Component } from 'react';

class TabStrip extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <ul className="nav nav-tabs mb-4">
                {
                    this.props.tabs.map(function (tab) {
                        return <li className='nav-item' key={tab.title}> <a className={tab.isActive ? 'nav-link active' : 'nav-link'} href="{tab.url}">{tab.title}</a></li>;
                    })
                }
            </ul>
        );
    }
}

export default TabStrip;