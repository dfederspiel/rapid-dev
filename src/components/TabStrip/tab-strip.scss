@import '../../styles/variables';
@import '../../styles/mixins';

.nav.nav-tabs {
    border-bottom-color: $primary;
    border-bottom-width: .2rem;
    font-size: 1.4rem;

    .nav-link {
        color: #B2B8BD;
        text-decoration:none;

        @include states {
            border-color: $primary;
        }
    }

    .nav-link.active {
        color: #FFF;
        border-color: $primary;
        background-color: $primary;
    }
}
