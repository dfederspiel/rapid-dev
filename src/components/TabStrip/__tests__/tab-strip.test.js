import React from 'react';
import TabStrip from '../tab-strip';
import renderer from 'react-test-renderer';

describe('the tab strip component', () => {

    const props = {
        tabs: [
            {
                title: 'Name',
                url: 'testUrl',
                isActive: true
            },
            {
                title: 'Address',
                url: 'anotherUrl',
                isActive: false
            }]
    };

    it('renders correctly', () => {
        const tree = renderer.create(
            <TabStrip
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});