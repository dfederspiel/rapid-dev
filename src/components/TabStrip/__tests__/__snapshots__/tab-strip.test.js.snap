// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the tab strip component renders correctly 1`] = `
<ul
  className="nav nav-tabs mb-4"
>
  <li
    className="nav-item"
  >
     
    <a
      className="nav-link active"
      href="{tab.url}"
    >
      Name
    </a>
  </li>
  <li
    className="nav-item"
  >
     
    <a
      className="nav-link"
      href="{tab.url}"
    >
      Address
    </a>
  </li>
</ul>
`;
